redwood-leak-detector/src/commonMain/kotlin/app/cash/redwood/leaks/LeakDetector.kt

```diff
@@ -27,7 +27,7 @@ public interface LeakDetector {
    *
    * This function is safe to call from any thread.
    */
-  public fun watchReference(reference: Any, name: String)
+  public fun watchReference(reference: Any, description: () -> String)
```

Comment:
Is doing this lazily going to encourage people to create leaks? I can imagine this lambda capturing stuff it aught not to

Reply:
Yeah I think this can change to a raw String. This predates me adding generated `toString()` to the actual references, so they're all constant strings now rather than dynamic ones.